biz.aQute.bndlib.tests/src/test/component/DSAnnotationTest.java

```diff
@@ -1160,6 +1160,86 @@ public static void testBinds() throws Exception {
 
 	}
 	
+	@Component
+	public static class CheckBinds13 {
+
+		@Reference
+		private void bindService(LogService l) {}
+
+		protected void unbindService(LogService l) {}
+
+		void updatedService(LogService l) {}
+
+		@Reference
+		private void bindSR(ServiceReference<LogService> l) {}
+
+		protected void unbindSR(ServiceReference<LogService> l) {}
+
+		void updatedSR(ServiceReference<LogService> l) {}
+
+		@Reference(service = LogService.class)
+		private void bindProps(Map<String,Object> l) {}
+
+		protected void unbindProps(Map<String,Object> l) {}
+
+		void updatedProps(Map<String,Object> l) {}
+
+		// @Reference
+		// private void bindSO(ComponentServiceObjects<LogService> l) {}
+		//
+		// protected void unbindSO(ComponentServiceObjects<LogService> l) {}
+		//
+		// void updatedSO(ComponentServiceObjects<LogService> l) {}
```

Comment:
Is this commented out because you did not have the updated component API present in the last commit? Can you uncomment now?